src/Microsoft.Health.Fhir.Liquid.Converter/Json/JsonProcessor.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
+using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Microsoft.Health.Fhir.Liquid.Converter.OutputProcessor;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Json
+{
+    public class JsonProcessor : BaseProcessor
+    {
+        private readonly JsonDataParser _dataParser = new JsonDataParser();
+
+        public JsonProcessor(ProcessorSettings processorSettings = null)
+            : base(processorSettings)
+        {
+        }
+
+        public override string Convert(string data, string rootTemplate, ITemplateProvider templateProvider, TraceInfo traceInfo = null)
```

Comment:
Looks like we can reuse the Convert logic for v2, ccda and json processors?

Reply:
How about add a new interface with a cancellation token for rendering

Reply:
> Looks like we can reuse the Convert logic for v2, ccda and json processors?

The conversion logics for HL7 v2, C-CDA and JSON are a bit different. HL7 v2 and C-CDA to FHIR conversion need to load different CodeSystem/ValueSet mappings in their context. HL7 v2 supports returning the debug information (unused segments, etc.) in the output. So the `Convert` function has different implementations for different input data types.

Reply:
> How about add a new interface with a cancellation token for rendering

Thanks, this is already supported in `BaseProcessor`.

Reply:
> > Looks like we can reuse the Convert logic for v2, ccda and json processors?
> 
> The conversion logics for HL7 v2, C-CDA and JSON are a bit different. HL7 v2 and C-CDA to FHIR conversion need to load different CodeSystem/ValueSet mappings in their context. HL7 v2 supports returning the debug information (unused segments, etc.) in the output. So the `Convert` function has different implementations for different input data types.

Yes. Since context and traceinfo are referenced in all interfaces. Looks we can keep a basic function that create context and do trace info collections in BaseProcessor and override the functions when needed (like in v2 and cda processor)?

Reply:
> > How about add a new interface with a cancellation token for rendering

Do we plan to add cancellation token in the core convert logic as well as in the interface? like [here](https://github.com/dotliquid/dotliquid/pull/406/files)

Reply:
Thanks for the suggestion! A base function is added in `BaseProcessor` . `BaseProcessor` would provide virtual methods `CreateContext` and `CreateTraceInfo` for derived processors to override.

As for cancellation token, we could write a new `CreateContext` for it when we update the dotliquid package.

Reply:
Sounds good. Thanks!